src/scss/05_objects/_button.scss

```diff
@@ -132,208 +178,272 @@ input[type='submit'].hds-button {
   padding: var(--spacing-s) var(--spacing-l);
 }
 
-/* no icons */
-
+// no icons
 .hds-button__label:only-child {
   margin: 0 var(--spacing-2-xs);
 }
 
-/* supplementary with right icon */
+input[type='button'].hds-button--small,
+input[type='reset'].hds-button--small,
+input[type='submit'].hds-button--small,
+.hds-button--small .hds-button__label {
+  line-height: var(--lineheight-s);
+  padding: var(--spacing-2-xs) var(--spacing-xs);
+}
 
+// supplementary with right icon
 .hds-button--supplementary .hds-button__label:first-child {
   padding-right: var(--spacing-3-xs);
 }
 
-/* supplementary with left icon */
+// no icons
+.hds-button--small .hds-button__label:only-child {
+  margin: 0 var(--spacing-xs);
+}
 
+// supplementary with left icon
 .hds-button--supplementary .hds-icon + .hds-button__label:last-child {
   padding-left: var(--spacing-3-xs);
 }
 
-/* supplementary with both icons */
-
+// supplementary with both icons
 .hds-button--supplementary .hds-icon + .hds-button__label:not(:last-child) {
   padding-left: var(--spacing-3-xs);
   padding-right: var(--spacing-3-xs);
 }
 
-/* ICONS */
+// SMALL
+.hds-button--small {
+  padding: 0;
+}
+
+input[type='button'].hds-button--small,
+input[type='reset'].hds-button--small,
+input[type='submit'].hds-button--small {
+  padding: var(--spacing-2-xs) var(--spacing-m);
```

Comment:
This is overriding styles set on line 191. Is this intentional?

Reply:
This part of our code tries to closely follow HDS codes. This part is in https://github.com/City-of-Helsinki/helsinki-design-system/blob/76ed6122f96cfe847b0ff94c13ea0afd9e7cc067/packages/core/src/components/button/button.css#L233C15-L233C15 and has a same problem. So kind of intentional on our end, but do not know how intentional is it is on their end.